packages/cli/src/commands/rewards/show.ts

```diff
@@ -0,0 +1,261 @@
+      const totalGroupVotes: { [key: string]: BigNumber } = {}
+      if (address) {
+        const voter = await election.getVoter(address, blockNumber)
+        for (const vote of voter.votes) {
+          const group: string = vote.group.toLowerCase()
+          addressVotes[group] = (addressVotes[group] || new BigNumber(0)).plus(vote.active)
+          totalGroupVotes[group] = (totalGroupVotes[group] || new BigNumber(0)).plus(
+            await election.getTotalVotesForGroup(group, blockNumber)
+          )
+        }
+      }
+
+      // voterRewardsEvent applies to address when voterRewardsEvent.group in addressVotes.
+      const epochVoterRewardsEvents = await election.getPastEvents(
+        'EpochRewardsDistributedToVoters',
+        {
+          fromBlock: blockNumber,
+          toBlock: blockNumber,
+        }
+      )
+
+      const voterRewardsValidatorGroupDetails: ValidatorGroup[] = await Promise.all(
+        epochVoterRewardsEvents.map((e: EventLog) =>
+          validators.getValidatorGroup(e.returnValues.group)
+        )
+      )
+
+      const epochVoterRewards = epochVoterRewardsEvents.map(
+        (e: EventLog, index: number): VoterReward => ({
+          name: voterRewardsValidatorGroupDetails[index].name,
+          group: e.returnValues.group,
+          value: e.returnValues.value,
+          blockNumber: e.blockNumber,
+          addressVotes: addressVotes[e.returnValues.group],
+          totalGroupVotes: totalGroupVotes[e.returnValues.group],
+        })
+      )
+
+      voterRewards = voterRewards.concat(
+        address
+          ? epochVoterRewards.filter((e: VoterReward) => e.group.toLowerCase() in addressVotes)
+          : epochVoterRewards
+      )
+
+      // validatorRewardEvent applies to address when validatorRewardEvent.validator is address.
+      const epochValidatorRewardsEvents = await validators.getPastEvents(
+        'ValidatorEpochPaymentDistributed',
+        {
+          fromBlock: blockNumber,
+          toBlock: blockNumber,
+        }
+      )
+
+      const epochValidatorDetails: Validator[] = await Promise.all(
+        epochValidatorRewardsEvents.map((e: EventLog) =>
+          validators.getValidator(e.returnValues.validator, blockNumber)
+        )
+      )
+
+      validatorRewards = validatorRewards.concat(
+        epochValidatorRewardsEvents.map(
+          (e: EventLog, index: number): ValidatorReward => ({
+            name: epochValidatorDetails[index].name,
+            validator: e.returnValues.validator,
+            validatorPayment: e.returnValues.validatorPayment,
+            validatorScore: epochValidatorDetails[index].score,
+            group: e.returnValues.group,
+            blockNumber: e.blockNumber,
+          })
+        )
+      )
+
+      // validatorRewardEvent applies to address when validatorRewardEvent.group is address.
+      const epochValidatorGroupDetails: ValidatorGroup[] = await Promise.all(
```

Comment:
use `concurrentMap()` since you might be making hundreds of concurrente request at the same time here